src/Microsoft.AspNetCore.TestHost/WebSocketClient.cs

```diff
@@ -87,6 +87,7 @@ private class RequestState : IDisposable, IHttpWebSocketFeature
         {
             private readonly IHttpApplication<Context> _application;
             private TaskCompletionSource<WebSocket> _clientWebSocketTcs;
+            private CancellationTokenRegistration _cancellationTokenRegistration;
```

Comment:
Well 'break this fix'... the difference is it might stay around until the next GC instead of until whenever RequestState gets disposed. Certainly whenever it gets disposed cancellation will break. I'm just not sure exactly how GC would work; if the client has a reference to a task linked to `_clientWebSocketTcs` would that mean RequestState would stay around? That'd mean it'd be OK to properly clean up both the registration and the tcs in RequestState.Dispose. Not that it matters much because no one will have an explicit reference to RequestState...it'll be disposed at GC time anyway.